File: week2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

float determine_grade(string text);

int main(void)
{
    string text = get_string("Text: ");
    float grade = determine_grade(text);

    if (grade > 16) printf("Grade 16+\n");
    else if (grade < 1) printf("Before Grade 1\n");
    else printf("Grade %.0f\n", grade);

    return 0;
}

/*
*   Counts letters, words and sentences
*/
float determine_grade(string text) {
    string grade;
    float letters=0, sentences=0, words=0;
    float index;
    for (int i=0, n=strlen(text); i < n; i++)
    {
        // Counts letters
        if (isalpha(text[i])) letters++;

        // Counts sentences based on amount of punctuation
        switch (text[i]) {
            case '.':
            case '!':
            case '?':
                sentences++;
                break;
            case ' ':
                words++;
                break;
        }
    }
    // Correct the number of words, since all of the times, the number of words is one more than the number of spaces
    words++;

    index = 0.0588 * (letters/words*100) - 0.296 * (sentences/words*100) - 15.8;
    printf("%f, %f, %f, %f\n", letters, words, sentences, index);
    return index;
}

File: week6/pset6/sentimental-readability/readability.py
# TODO
from cs50 import get_string

def main():
    text = get_string("Text: ")
    words = text.count(" ") +1
    letters = 0
    sentences = 0
    for char in text:
        if char.isalpha():
            letters += 1
        if char in ['.', '!', '?']:
            sentences += 1

    grade = 0.0588 * (letters/words*100) - 0.296 * (sentences/words*100) - 15.8;
    print(letters, words, sentences, grade)
    if grade > 16: print("Grade 16+")
    elif grade < 1: print("Before Grade 1")
    else: print(f"Grade {grade:.0f}")


main()
